ludwig/features/audio_feature.py

```diff
@@ -148,19 +146,23 @@ def _process_in_memory(
         audio_file_length_limit_in_s,
         backend,
     ):
-        def read_audio_file(path):
-            return read_audio(path, src_path)
 
         df_engine = backend.df_engine
-        raw_audio = df_engine.map_objects(column, read_audio_file)
+        raw_audio = backend.read_binary_files(column, map_fn=read_audio_if_bytes_obj)
+
+        def is_torch_audio_tuple(audio):
+            if isinstance(audio, tuple):
+                if isinstance(audio[0], torch.Tensor) and isinstance(audio[1], int):
```

Comment:
Do we want to check tuple length as well to be safe?

Reply:
Sure! Added.